src/Symfony/Component/Translation/Tests/DefaultParametersTranslatorTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Translation\DefaultParametersTranslator;
+use Symfony\Component\Translation\Loader\ArrayLoader;
+use Symfony\Component\Translation\Translator;
+
+class DefaultParametersTranslatorTest extends TestCase
+{
+    public function testTrans()
+    {
+        $translator = new Translator('en');
+        $translator->addLoader('array', new ArrayLoader());
+        $translator->addResource('array', ['test' => 'Welcome %name%!'], 'en');
+
+        $defaultParametersTranslator = new DefaultParametersTranslator($translator);
+        $defaultParametersTranslator->addDefaultParameter('%name%', 'Global name');
+
+        $this->assertSame('Welcome Global name!', $defaultParametersTranslator->trans('test'));
+        $this->assertSame('Welcome Name!', $defaultParametersTranslator->trans('test', ['%name%' => 'Name']));
+    }
+
```

Comment:
Could you add more tests to cover both Message format and Intl format please?

Reply:
I added some tests with a different locale and with the ICU format. Is it OK for you ?